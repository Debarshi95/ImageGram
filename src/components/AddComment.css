.addcomment__root {
  display: flex;
  width: 100%;
  padding: 1rem;
  /* align-items: center; */
}

.addcomment__root > textarea {
  border: 0;
  outline: none;
  padding: 0 14px;
  flex: 1;
  font-family: inherit;
  max-height: 70px;
  resize: none;
}

.addcomment__root > button {
  border: 0;
  outline: none;
  color: blueviolet;
  background: transparent;
  font-weight: 600;
  cursor: pointer;
}
