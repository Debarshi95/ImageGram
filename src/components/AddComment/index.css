.addcomment__root {
  display: flex;
  width: 100%;
  align-items: flex-start;
  height: 3rem;
}

.addcomment__root > textarea {
  border: 0;
  outline: none;
  flex: 1;
  font-family: inherit;
  resize: none;
  padding: 0.75rem 0.65rem;
  height: 100%;
}
.addcomment__root > textarea::-webkit-scrollbar {
  display: none;
}
.addcomment__root > button {
  border: 0;
  outline: none;
  color: blueviolet;
  background: transparent;
  font-weight: 600;
  cursor: pointer;
  text-transform: uppercase;
  font-family: inherit;
  font-weight: 600;
  padding: 0.25rem 1.4rem;
  border-left: 0.1rem solid #bdbdbd;
  height: 100%;
}
