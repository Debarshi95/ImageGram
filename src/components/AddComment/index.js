import React from "react";
import "./index.css";
import { addCommentToPost } from "../../firebase";
import { useAuth } from "../../provider/AuthProvider";

function AddComment({ imageId }) {
  const [comment, setComment] = React.useState("");
  const { user } = useAuth();

  const handleComment = async (e) => {
    e.preventDefault();
    await addCommentToPost(imageId, user.uid, comment);
    setComment("");
  };

  return (
    <div className="addcomment__root">
      <textarea
        type="text"
        name="comment"
        placeholder="Add a comment.."
        autoComplete="off"
        aria-label="Add a comment.."
        onChange={({ target }) => setComment(target.value)}
        value={comment}
      ></textarea>
      <button type="click" onClick={handleComment} disabled={comment === ""}>
        Post
      </button>
    </div>
  );
}

export default AddComment;
