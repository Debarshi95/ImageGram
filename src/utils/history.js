/* eslint-disable import/prefer-default-export */
import { createBrowserHistory, createMemoryHistory } from 'history';

const isTest = () => {
  return process.env.NODE_ENV === 'test';
};

export const history = isTest()
  ? createMemoryHistory({ initialEntries: ['/'] })
  : createBrowserHistory();
