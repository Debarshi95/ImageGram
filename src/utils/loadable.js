import React from 'react';

export default function loadable(importFunc) {
  return React.lazy(importFunc);
}
