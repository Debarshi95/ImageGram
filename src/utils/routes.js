export default {
  home: {
    path: '/',
    exact: true,
  },
  signIn: {
    path: '/signin',
    exact: true,
  },
  signUp: {
    path: '/signup',
    exact: true,
  },
  upload: {
    path: '/upload',
    exact: true,
    isProtected: true,
  },
};
