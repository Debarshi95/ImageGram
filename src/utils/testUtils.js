import { render } from '@testing-library/react';
import { AuthContext } from '../provider/AuthProvider';

export const timeout = (delay) => new Promise((res) => setTimeout(res, delay));
export const renderWithAuthProvider = (Comp, value = {}) =>
  render(<AuthContext.Provider value={value}>{Comp}</AuthContext.Provider>);
