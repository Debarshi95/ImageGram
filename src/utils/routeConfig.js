import loadable from './loadable';
import routes from './routes';

export default {
  home: {
    component: loadable(() => import('../pages/Home')),
    ...routes.home,
  },
  signIn: {
    component: loadable(() => import('../pages/SignIn')),
    ...routes.signIn,
  },
  signUp: {
    component: loadable(() => import('../pages/SignUp')),
    ...routes.signUp,
  },
  upload: {
    component: loadable(() => import('../pages/Upload')),
    ...routes.upload,
  },
  notFound: {
    component: loadable(() => import('../pages/NotFound')),
  },
};
