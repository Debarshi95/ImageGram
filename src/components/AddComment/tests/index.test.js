import { fireEvent } from '@testing-library/react';
import { renderWithAuthProvider } from '../../../utils/testUtils';
import AddComment from '../index';

describe('<AddComment/> tests', () => {
  let contextValue;
  beforeEach(() => {
    contextValue = { user: null };
  });
  it('should render and match the snapshot', () => {
    const imageId = 123;
    const { baseElement } = renderWithAuthProvider(<AddComment imageId={imageId} />, contextValue);
    expect(baseElement).toMatchSnapshot();
  });

  it('should change the value on change', async () => {
    contextValue = { user: { id: 1, name: 'Dev' } };
    const imageId = 123;
    const { findByRole } = renderWithAuthProvider(<AddComment imageId={imageId} />, contextValue);

    const textArea = await findByRole('textbox');
    fireEvent.change(textArea, { target: { value: 'abc' } });

    expect(textArea.value).toEqual('abc');
  });

  // it("should set the value to '' and prevent default behavior on click", async () => {
  //   contextValue = { user: { id: 1, name: "Dev" } };
  //   const imageId = 123;
  //   const { findByRole } = renderWithAuthProvider(
  //     <AddComment imageId={imageId} />,
  //     contextValue
  //   );

  //   const textArea = await findByRole("textbox");
  //   const btn = await findByRole("button");

  //   const event = { preventDefault: jest.fn() };
  //   fireEvent.change(textArea, { target: { value: "xyz" } });

  //   fireEvent.click(btn, event);

  //   await timeout(500);
  //   expect(event.preventDefault).toBeCalled();
  //   expect(textArea.value).toEqual("");
  // });
});
