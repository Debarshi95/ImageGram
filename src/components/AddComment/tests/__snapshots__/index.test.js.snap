// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddComment/> tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="addcomment__root"
    >
      <textarea
        aria-label="Add a comment.."
        autocomplete="off"
        name="comment"
        placeholder="Add a comment.."
        type="text"
      />
      <button
        disabled=""
        type="click"
      >
        Post
      </button>
    </div>
  </div>
</body>
`;
